compiler/noirc_driver/src/lib.rs

```diff
@@ -107,6 +108,24 @@ fn parse_expression_width(input: &str) -> Result<ExpressionWidth, std::io::Error
     }
 }
 
+#[derive(Debug, Error)]
+pub enum CompileError {
+    #[error(transparent)]
+    MonomorphizationError(#[from] MonomorphizationError),
+
+    #[error(transparent)]
+    RuntimeError(#[from] RuntimeError),
+}
+
+impl From<CompileError> for FileDiagnostic {
+    fn from(error: CompileError) -> FileDiagnostic {
+        match error {
+            CompileError::RuntimeError(err) => err.into(),
+            CompileError::MonomorphizationError(err) => err.into(),
```

Comment:
We will want some way to distinguish that this is an internal error similar to how we do for `InternalError` inside of SSA. cc @jfecher

Reply:
Just when we convert an internal error to a diagnositc we state that we have an internal consistency error and link to our repo that an issue should be opened. A monomorphization bug should be handled in the same way as the pass is not expected to fail.

Reply:
Not all of the errors in the monomorphisation pass are internal consistency errors though. For example in https://github.com/noir-lang/noir/issues/4407 it's purely down to the input being well defined.

Reply:
They are all errors in the compiler though and should be stated as such. What we do for internal consistency errors was simply an example. We can state "Internal Monomorphization Errors: Please open issue here..." or something similar.

Reply:
Shall we chat about this on the call after scrum?

Reply:
Let's just talk during/post scrum so @jfecher can comment as well.